hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/rsql/RSQLUtilityTest.java

```diff
@@ -99,6 +100,43 @@ TenantConfigurationManagementHolder tenantConfigurationManagementHolder() {
     private static final TenantConfigurationValue<String> TEST_POLLING_OVERDUE_TIME_INTERVAL = TenantConfigurationValue
             .<String> builder().value("00:07:37").build();
 
+    @Test(expected = RSQLParameterUnsupportedFieldException.class)
+    @Description("Testing throwing exception in case of not allowed RSQL key")
+    public void rsqlUnsupportedFieldExceptionTest(){
+        final String rsql = "wrongfield == abcd";
+        RSQLUtility.isValid(rsql);
+    }
+
+    @Test(expected = RSQLParameterUnsupportedFieldException.class)
+    @Description("Testing exception in case of not allowed subkey")
+    public void rsqlUnsupportedSubkeyThrowException(){
+        final String rsql = "ASSIGNEDDS.unsupported == abcd and ASSIGNEDDS.version == 0123";
+        RSQLUtility.isValid(rsql);
+    }
+
+    @Test
+    @Description("Testing allowed RSQL keys based on TargetFields.class")
+    public void rsqlValidTargetFields(){
+        final String rsql1 = "ID == '0123' and NAME == abcd and DESCRIPTION == absd" +
```

Comment:
In case you would really like to test the Target fields explicitly you could extend the RSQLTargetFieldTest or introduce a new one e.g. RSQLTargetFieldValidationTest

Reply:
Ended up creating a new RSQLTargetFieldValidationTest class since it doesn't really fit into the existing RSQLTargetFieldTest, which operates heavily on JPA/Repository layer.